src/PollingBlockTracker.test.ts

```diff
@@ -185,6 +185,112 @@ describe('PollingBlockTracker', () => {
       );
     });
 
+    it('should return a promise that rejects if the request for the block number fails and the block tracker is then stopped', async () => {
+      recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                error: 'boom',
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          const latestBlockPromise = blockTracker.getLatestBlock();
+
+          expect(blockTracker.isRunning()).toBe(true);
+          await blockTracker.destroy();
+          await expect(latestBlockPromise).rejects.toThrow(
+            'Block tracker destroyed',
+          );
+          expect(blockTracker.isRunning()).toBe(false);
+        },
+      );
+    });
+
+    it('should not retry failed requests after the block tracker is stopped', async () => {
+      recordCallsToSetTimeout({ numAutomaticCalls: 1 });
```

Comment:
```suggestion
      recordCallsToSetTimeout({ numAutomaticCalls: 2 });
```
If I understand correctly how this mocked `setTimeout` works, by making two automatic calls we ensure that the timer ticks twice while the provider is called only once